sympy/assumptions/satask.py

```diff
@@ -332,7 +334,12 @@ def get_all_relevant_facts(proposition, assumptions, context,
 
     if use_known_facts:
         known_facts_CNF = CNF()
-        known_facts_CNF.add_clauses(get_all_known_facts())
+        for expr in all_exprs:
+            if expr.kind == MatrixKind(NumberKind):
+                known_facts_CNF.add_clauses(get_all_known_matrix_facts())
+            elif expr.kind == NumberKind:
+                known_facts_CNF.add_clauses(get_all_known_number_facts())
```

Comment:
Since the kind system is incomplete it will often return `UndefinedKind`. I suggest that in that case it should be treated as `NumberKind`.